### PR TITLE
3 - products and user auth web

### PR DESCRIPTION
**Pull request by WesJourdan**
- Add `productRTK` to show an example of how redux toolkit query works. Just a proof of concept, we probably won't use it.
- Add `product` with Yote inspired actions/reducers using new redux store stuff.
- Add `productService` to give us a simple hooks based API so we don't have to deal with actions/reducers in each component.
- Add `authStore` and `authService` to take care of user sessions.
- Port over a lot of the global stuff from original Yote updated with new syntax (YTRoute, DefaultLayout, etc...).

NOTE: Can ignore `server-old, web-old` and to some extent everything in the `productRTK` folder. Main things here are in `product` and `user` folders.

### CODE REVIEW

File: web-experimental/public/manifest.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
```
Comment (grantfowler) [inline web-experimental/public/manifest.json:2]: update!

File: web-experimental/src/global/components/base/Button.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    , link
+    , size = 'md'
+    , skin = 'primary'
+    , type = 'button'
+  } = props
+
+  const baseClasses = "inline-flex items-center border border-transparent font-medium shadow-sm focus:outline-none focus:ring-2 focus:ring-offset-2"
+
+  const colorClasses = (
+    skin === 'primary' ? "text-white bg-indigo-600 hover:bg-indigo-700 focus:ring-indigo-500"
```
Comment (grantfowler) [inline web-experimental/src/global/components/base/Button.jsx:22]: this component feels kind of gross. its not what i care about for this pull request though.

File: web-experimental/src/resources/counter/README.md [hunk 3]
```diff
@@ -0,0 +1 @@
+All of the 'counter' stuff comes from the redux cra template. Here for reference.
```
Comment (grantfowler) [inline web-experimental/src/resources/counter/README.md:1]: was going to leave a comment until i saw this. assuming we would remove once we feel more comfortable with it all?

**Review by grantfowler — APPROVED** [review]
ok so I have TONS of questions - but I think a lot of them will be best answered by actually USING this. since this is still technically an experimental branch, i'm going to approve and merge.

File: web-experimental/src/global/components/base/Button.jsx (hunk 2 above)
Comment (WesJourdan) [reply web-experimental/src/global/components/base/Button.jsx:22]: How so? It's so easy to use.

**Pull request merged**